src/node_crypto.cc

```diff
@@ -4317,8 +4317,9 @@ void CipherBase::Update(const FunctionCallbackInfo<Value>& args) {
   // Only copy the data if we have to, because it's a string
   if (args[0]->IsString()) {
     StringBytes::InlineDecoder decoder;
-    if (!decoder.Decode(env, args[0].As<String>(), args[1], UTF8)
-             .FromMaybe(false))
+    enum encoding enc = ParseEncoding(env->isolate(), args[1], UTF8);
+
+    if (!decoder.Decode(env, args[0].As<String>(), enc).FromMaybe(false))
```

Comment:
```suggestion
    if (decoder.Decode(env, args[0].As<String>(), enc).IsEmpty())
```

Reply:
I tried this approach but `IsEmpty()` doesn't seem to be a method of `Maybe`. Another option may be to change `v8::Maybe<bool> Decode(...)` to `bool Decode(...)` and end up with `if (!decoder.Decode(env, args[0].As<String>(), enc)) {...}`.

Reply:
I think Joyee meant `.IsNothing()` rather than `.IsEmpty()`? I always mix these two up too…

Reply:
Yes, I meant IsNothing(). Sorry about the confusion.

Reply:
Fixed.